src/perf/lib/Tcp.cpp

```diff
@@ -192,16 +192,29 @@ TcpWorker::TcpWorker()
 TcpWorker::~TcpWorker()
 {
     CXPLAT_FRE_ASSERT(!Connections);
-    if (Initialized) {
+    if (Initialized && !IsExternal) {
         CxPlatThreadDelete(&Thread);
     }
     CxPlatDispatchLockUninitialize(&Lock);
     CxPlatEventUninitialize(WakeEvent);
 }
 
-bool TcpWorker::Initialize(TcpEngine* _Engine)
+bool TcpWorker::Initialize(TcpEngine* _Engine, uint16_t AssignedCPU)
 {
     Engine = _Engine;
+    ExecutionContext.Callback = DoWork;
+    ExecutionContext.Context = this;
+    ExecutionContext.Ready = TRUE;
+    ExecutionContext.NextTimeUs = 0;
+    PartitionIndex = AssignedCPU;
+
+    if (Engine->TcpExecutionProfile == TCP_EXECUTION_PROFILE_LOW_LATENCY) {
+        CxPlatAddExecutionContext(&ExecutionContext, AssignedCPU);
+        Initialized = true;
+        IsExternal = true;
+        return true;
+    }
+
     CXPLAT_THREAD_CONFIG Config = { 0, 0, "TcpPerfWorker", WorkerThread, this };
```

Comment:
Also BTW, in the `maxtput` case, why do set the ideal processor to always be processor 0 here?

Reply:
Probably a bug 😄

Reply:
Set it to AssignedCPU